sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -405,6 +413,8 @@ abstract class StreamExecution(
           new QueryTerminatedEvent(id, runId, exception.map(_.cause).map(Utils.exceptionString),
             errorClassOpt))
 
+        loggingThreadContext.close()
```

Comment:
let's check if `loggingThreadContext` is null.

Reply:
Ah yea we should definitely do that, thank you!